Notice/Event.swift

```diff
@@ -0,0 +1,32 @@
+public protocol Event {
+    typealias ValueType
+    typealias HandlerType
+
+    var value: ValueType { get }
+    func handle(handler: HandlerType)
+}
+
+public struct NewEvent<T>: Event {
+    public typealias HandlerType = T -> Void
+    public let value: T
+
+    public func handle(handler: HandlerType) {
```

Comment:
If you change the signature of the argument here from `handler: HandlerType` to `handler: T -> Void`, you can get rid of the `typealias` above.